### PR TITLE
glue_connection - Avoid converting connection_parameter keys to lowercase

### PR DESCRIPTION
**Pull request by ichekaldin**
##### SUMMARY

This is a follow to my own PR [#503](https://github.com/ansible-collections/community.aws/pull/503).

This is a cosmetic change that prevents converting keys in connection_parameters dict to lowercase.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

aws_glue_connection

##### ADDITIONAL INFORMATION

As an example, this:

```
- community.aws.glue_connection:
    name: test-connection
    connection_parameters:
      JDBC_ENFORCE_SSL: "false"
    ...
```

is a valid value, while this:

```
- community.aws.glue_connection:
    name: test-connection
    connection_parameters:
      jdbc_enforce_ssl: "false"
    ...
```

is not.

This PR simply aligns the module output to the expected input.

### CODE REVIEW

**Comment by jillr** [comment]
Thanks for this patch @ichekaldin. However this would be a breaking change, and would need a deprecation period where we would maintain both formats.  

I haven't really used Glue myself, but it looks like the AWS docs use the upper-cased values https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue.html#Glue.Client.get_connection.  I'd like to hear what other Glue users think about this change.  Can the lower-cases values be used as the input for another Glue task?

**Comment by ichekaldin** [comment]
@jillr from your point of view - is it worth working on this PR further?

**Comment by tremble** [comment]
@ichekaldin apologies for the delay here.  In my opinion, this is the correct change for the long run, the question is how to 'deprecate' the old key.  We're currently discussing a general cleanup of return values (https://github.com/ansible-collections/amazon.aws/pull/865)

To ensure people have a deprecation cycle available to them the consensus seems to be
- Add a new key with the new value
- Add a deprecation notice that the original key will be updated/removed after 2 years

As such, I'd suggest here is that we add a new return value `raw_connection_parameters` and a deprecation notice.  We'll then just have to wait for the deprecation cycle to complete before changing the original `connection_parameters` value.

File: changelogs/fragments/518-aws_glue_connection-do-not-lowercase-connection-parameters.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- aws_glue_connection - avoid converting connection_parameters to lowercase (https://github.com/ansible-collections/community.aws/pull/518).
```
Comment (tremble) [inline changelogs/fragments/518-aws_glue_connection-do-not-lowercase-connection-parameters.yml:2]:
```suggestion
- aws_glue_connection - added new ``raw_connection_parameters`` return key which doesn't snake case the connection parameters (https://github.com/ansible-collections/community.aws/pull/518).
deprecated_features:
- aws_glue_connection - the ``connection_parameters`` return key has been deprecated and will be removed in a release after 2024-06-01, it is being replaced by the ``raw_connection_parameters`` key (https://github.com/ansible-collections/community.aws/pull/518).
```

File: plugins/modules/glue_connection.py [hunk 2]
```diff
@@ -309,7 +314,9 @@ def create_or_update_glue_connection(connection, connection_ec2, module, glue_co
     if changed and not module.check_mode:
         glue_connection = _await_glue_connection(connection, module)
 
-    module.exit_json(changed=changed, **camel_dict_to_snake_dict(glue_connection or {}))
+    glue_connection['RawConnectionProperties'] = glue_connection['ConnectionProperties']
+
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(glue_connection or {}, ignore_list=['ConnectionProperties']))
```
Comment (tremble) [inline plugins/modules/glue_connection.py:319]:
```suggestion
    module.exit_json(changed=changed, **camel_dict_to_snake_dict(glue_connection or {}, ignore_list=['RawConnectionProperties']))
```

**Review by tremble — CHANGES_REQUESTED** [review]

**Comment by ichekaldin** [comment]
@tremble Thank you for all the suggestions and edits!

**Review by tremble — APPROVED** [review]

**Pull request merged**